circuits/batch_update_state_tree.circom

```diff
@@ -0,0 +1,124 @@
+include "./update_state_tree.circom";
+
+template BatchUpdateStateTree(
+  depth,
+  vote_options_tree_depth,
+  batch_size
+) {
+  // params:
+  //    depth: the depth of the state tree and the command tree
+  //    vote_options_tree_depth: depth of the vote tree
+  //    batch_size: that we're processing
+
+  // Output: New state tree root
+  signal output root;
+
+  // Input(s)
+  signal input coordinator_public_key[2];
+
+  var message_length = 11;
+  signal input message[batch_size][message_length];
+
+  // Select vote option index's weight
+  // (a.k.a the raw value of the leaf pre-hash)
+  signal private input vote_options_leaf_raw[batch_size];
+
+  // Vote options tree root (supplied by coordinator)
+  signal private input vote_options_tree_root[batch_size];
+  signal private input vote_options_tree_path_elements[batch_size][vote_options_tree_depth];
+  signal private input vote_options_tree_path_index[batch_size][vote_options_tree_depth];
+  signal input vote_options_max_leaf_index;
+
+  // Message tree
+  signal input msg_tree_root;
+  signal input msg_tree_path_elements[batch_size][depth];
+  signal input msg_tree_path_index[batch_size][depth];
```

Comment:
How about instead of heaving `msg_tree_path_index`, this would be calculated?
We'd have "initial index" and then we continue one by one until batch size? We'd need to then transform each index into binary, but that feels better than having all of these public inputs

Also, can `msg_tree_path_elements` be private inputs?

Reply:
Regarding `msg_tree_path_elements`, why does it need to be private inputs? As the msg tree can be reconstructed by anyone

Reply:
Yep, it's not for privacy! It's for having less public inputs if not needed. That's pretty expensive on chain